packages/create-cloudflare/e2e-tests/helpers.ts

```diff
@@ -340,36 +347,38 @@ const getLogPath = (opts: { experimental: boolean }, suite: Suite) => {
 	);
 };
 
-const normalizeTestName = (ctx: TaskContext) => {
-	const baseName = ctx.task.name
+const normalizeTestName = (task: Test) => {
+	const baseName = task.name
 		.toLowerCase()
 		.replace(/\s+/g, "_") // replace any whitespace with `_`
 		.replace(/\W/g, ""); // strip special characters
 
 	// Ensure that each retry gets its own log file
-	const retryCount = ctx.task.result?.retryCount ?? 0;
+	const retryCount = task.result?.retryCount ?? 0;
 	const suffix = retryCount > 0 ? `_${retryCount}` : "";
 	return baseName + suffix;
 };
 
-export const testProjectDir = (suite: string) => {
+export const testProjectDir = (suite: string, test: string) => {
 	const tmpDirPath =
 		process.env.E2E_PROJECT_PATH ??
 		realpathSync(mkdtempSync(path.join(tmpdir(), `c3-tests-${suite}`)));
 
 	const randomSuffix = crypto.randomBytes(4).toString("hex");
 	const baseProjectName = `${C3_E2E_PREFIX}${randomSuffix}`;
 
-	const getName = (suffix: string) => `${baseProjectName}-${suffix}`;
-	const getPath = (suffix: string) => path.join(tmpDirPath, getName(suffix));
-	const clean = (suffix: string) => {
+	const getName = () =>
+		// Worker project names cannot be longer than 58 characters
+		`${baseProjectName}-${test.substring(0, 57 - baseProjectName.length)}`;
```

Comment:
If 2 tests have the same prefix, this will cause them the clash. How likely is that to happen?

Reply:
The `baseProjectName` already has a random suffix, which makes these unique in general.

Reply:
Ah ok I assumed projectName was shared across the suite but I see now it's not